<launch>

  <!-- Configurations -->
  
  <!-- launch_kinect2 -->
  <arg name="launch_kinect2" default="true" doc="true if you want to include openni2_kinect2, this should be set to true or dynamic reconfigure wont load (bug)" />
  
  <!-- run_reconfigure:  If true runs rqt_reconfigure gui for dynamic reconfiguration of parameters.-->
  <arg name="run_reconfigure" default="true" doc="If true runs rqt_reconfigure gui for dynamic reconfiguration of parameters." />
  
  <!-- run_rviz:  If true runs rviz gui.-->
  <arg name="run_rviz" default="true" doc="If true runs rviz gui with config stored under cfg directory."/>
  
  <!-- load:  Load a preset of parameters from yaml file named as this argument (yaml files must be located into params subfolder).-->
  <arg name="load" default="default" doc="Load a preset of parameters from yaml file named as this argument (yaml files must be located into params subfolder of this node)."/>

  <!-- Implementation -->

  <group if="$(arg launch_kinect2)">
    <!-- Launch kinect2 with custom driver parameters -->
    <include file="$(find kinect2_bridge)/launch/kinect2_bridge.launch">  
      <arg name="publish_tf"             value="true" />
      <arg name="worker_threads"         value="4" />
    </include>
  </group>

  <rosparam command="load" file="$(find pacman_vision)/params/$(arg load).yaml"/>

	<!-- pacman vision node  -->
	<node name="pacman_vision" pkg="pacman_vision" type="pacman_vision" output="screen" />

  <!-- rviz -->
  <group if="$(arg run_rviz)">
	  <node name="rviz_pacman_vision" pkg="rviz" type="rviz" respawn="false" args="-d $(find pacman_vision)/cfg/config.rviz" output="screen"/>
  </group>
  
  <!-- rqt_reconfigure -->
  <group if="$(arg run_reconfigure)">
    <node name="dynamic_reconfigure_pacman_vision" pkg="rqt_reconfigure" type="rqt_reconfigure" respawn="false"/> 
  </group>
    
</launch>
